// ngx-mail-sender/projects/ngx-mail-sender/src/lib/ngx-mail-sender.component.ts
import { HttpClient } from '@angular/common/http';
import { Component } from '@angular/core';
import { Observable } from 'rxjs';
import { NgxMailSenderService } from './ngx-mail-sender.service';

@Component({
  selector: 'lib-ngx-mail-sender',
  template: `
     <p>
       ngx-mail-sender works!
     </p>
  `,
  styles: ``
 })
export class NgxMailSenderComponent {
 emailData = {
    to: '',
    subject: '',
    text: '',
    host:'',
    authUser:'',
    authPass:'',
    fromTitle:'',
   };

  constructor(private http:HttpClient,private ngxMailSenderService:NgxMailSenderService ){}


/**
 * Sends an email using the provided email data.
 *
 * @param {Object} emailData - The email data object.
 * @param {string} emailData.to - The recipient's email address.
 * @param {string} emailData.subject - The subject of the email.
 * @param {string} emailData.text - The body text of the email.
 * @param {string} emailData.host - The SMTP host for sending the email. For Gmail, use 'smtp.gmail.com'.
 * @param {string} emailData.authUser - The Gmail account email address from which the email will be sent.
 * @param {string} emailData.authPass - The password for the Gmail account. If 2-factor authentication is enabled, use an App Password generated by Google.
 * @param {string} emailData.fromTitle - The title for the sender's email address.
 *
 * @example
 * const emailData = {
 *   to: 'recipient@example.com',
 *   subject: 'Hello from MyApp',
 *   text: 'This is a test email.',
 *   host: 'smtp.gmail.com',
 *   authUser: 'user@gmail.com',
 *   authPass: 'your-app-password', // Use an Google app  Password of 2-factor authentication is enabled
 *   fromTitle: 'MyApp Support'
 * };
 *
 * sendEmail(emailData);
 */
sendEmail(emailData:any){
  this.ngxMailSenderService.sendMail(emailData).subscribe(
     response => console.log('Email sent successfully', response),
     error => console.error('Error sending email', error)
  );
 }

}
